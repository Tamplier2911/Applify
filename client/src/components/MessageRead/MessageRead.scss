.message-read {
  display: grid;

  &__header {
    display: grid;
    grid-template-columns: 1fr min-content;
  }

  &__name {
    order: 1;
  }
  &__email {
    order: 3;
  }

  &__portrait {
    &--wrap {
      width: 10rem;
      height: 10rem;
    }

    &--img {
      width: 100%;
      height: 100%;
      display: block;
      object-fit: cover;
    }
  }

  &__body {
  }
  &__message {
  }
  &__date {
    &--itself {
    }
  }
  &__controlls {
  }
}
