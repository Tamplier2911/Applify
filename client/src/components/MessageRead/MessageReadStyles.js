import styled, { css } from "styled-components";

const sharedMessageReadStyles = css`
  width: 50rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  @media only screen and (max-width: 720px) {
    width: 35rem;
  }

  @media only screen and (max-width: 425px) {
    width: 25rem;
  }

  @media only screen and (max-width: 300px) {
    width: 20rem;
  }
`;

export const MessageReadContainer = styled.div`
  display: grid;
  grid-row-gap: 2rem;
  font-family: var(--font-title);

  margin-bottom: 4rem;
`;

export const MessageReadHeader = styled.div`
  display: grid;
  grid-template-columns: 1fr min-content;

  @media only screen and (max-width: 425px) {
    grid-template-columns: 1fr;
    justify-items: center;
    grid-row-gap: 1rem;
  }
`;

export const MessageReadCredentials = styled.div`
  display: grid;
  grid-auto-rows: min-content;
  align-content: center;

  @media only screen and (max-width: 425px) {
    order: 3;
    text-align: center;
  }
`;

export const MessageReadName = styled.div`
  ${sharedMessageReadStyles}
  order: 1;
  font-size: 2rem;
  font-weight: bold;
`;

export const MessageReadEmail = styled.div`
  ${sharedMessageReadStyles}
  order: 3;
  font-size: 1.8rem;
`;

export const MessageReadPortrait = styled.div`
  order: 2;
`;

export const MessageReadWrapper = styled.div`
  width: 10rem;
  height: 10rem;
  overflow: hidden;
  border-radius: 50%;
  box-shadow: 0 0 0.8rem var(--cl-primary);
`;

export const MessageReadImg = styled.img`
  width: 100%;
  height: 100%;
  display: block;
  object-fit: cover;
`;

export const MessageReadBody = styled.div``;

export const MessageReadMessage = styled.div`
  font-size: 1.7rem;
`;

export const MessageReadDateWrap = styled.div`
  display: grid;
  justify-items: end;
`;

export const MessageReadDate = styled.div`
  color: var(--cl-darkest-grey);
`;

export const MessageReadControlls = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-column-gap: 2rem;
`;

export const MessageReadNotFound = styled.div`
  font-size: 1.8rem;
  font-family: var(--font-title);
`;
