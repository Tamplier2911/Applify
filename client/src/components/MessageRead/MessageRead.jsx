import "./MessageRead.scss";
import React from "react";
import { withRouter } from "react-router-dom";

// redux
import { connect } from "react-redux";
import { selectMessage } from "../../redux/messages/messages.selectors";

// components
import GetBack from "../GetBack/GetBack";

// JS Rendering CSS
import {} from "./MessageReadStyles";

const MessageRead = ({ messageObject }) => {
  const { createdAt, name, email, message, from } = messageObject;
  // FIX IMAGE LOADING ALGORITHM
  // ISSUES IS /profile/ find way to rid of profile
  //                                    \/      \/
  // https://applify-s.herokuapp.com/profile/messages/api/uploads/images/users/user-5e6e618672e9151d503701ed-1584642619899.jpeg
  // required href - https://applify-s.herokuapp.com/api/...
  let locationHref = window.location.href.split("/");
  locationHref = "".concat(locationHref[0], "//", locationHref[2], "/");

  let photo = "";
  if (from) {
    photo = from.photo;
  } else {
    photo = "uploads/images/users/default.jpg";
  }

  const date = new Date(createdAt).toLocaleString("en-us", {
    day: "numeric",
    month: "long",
    year: "numeric"
  });

  let image = "";
  if (process.env.NODE_ENV === "development" && photo) {
    image = `${process.env.REACT_APP_SERVE_IMAGE_DEV + "api/" + photo}`;
  } else if (photo) {
    image = `${process.env.REACT_APP_SERVE_IMAGE_PROD + "api/" + photo}`;
  }

  return (
    <div className="message-read">
      <div className="message-read__header">
        <div className="message-read__name">Name: {name}</div>
        <div className="message-read__hemail">Email: {email}</div>
        <div className="message-read__portrait">
          <div className="message-read__portrait--wrap">
            <img
              alt="happy user"
              src={image}
              className="message-read__portrait--img"
            ></img>
          </div>
        </div>
      </div>
      <div className="message-read__body">
        <div className="message-read__message">Message: {message}</div>
      </div>
      <div className="message-read__date">
        <div className="message-read__date--itself">Date: {date}</div>
      </div>
      <div className="message-read__date--controlls">
        <div>BTN</div>
        <div>BTN</div>
      </div>
      <GetBack path={"/profile/messages"} />
    </div>
  );
};

const mapStateToProps = (state, ownProps) => ({
  messageObject: selectMessage(ownProps.match.params.id)(state)
});

export default withRouter(connect(mapStateToProps)(MessageRead));
