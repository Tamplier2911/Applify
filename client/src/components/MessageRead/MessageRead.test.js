import React from "react";

// component
import MessageRead from "./MessageRead";

// utils
import {
  setShallowWRC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<MessageRead />", () => {
  const initialState = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
    messages: {
      allMessages: {
        sdk123: {
          _id: "sdk123",
          from: {},
          createdAt: Date.now(),
          name: "Test Message",
          email: "test@email.com",
          message: "To be tested!",
        },
      },
      isLoading: false,
      errorMessage: null,
    },
  };
  const store = storeFactory(initialState);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowWRC(
      <MessageRead store={store} testMessageId={"sdk123"} />
    );
  });

  it("renders without an error", () => {
    const messageRead = findByTestAttr(wrapper, "message-read");
    expect(messageRead.length).toBe(1);
  });

  it("renders children - message body", () => {
    const messageReadBody = findByTestAttr(wrapper, "message-read-body");
    expect(messageReadBody.length).toBe(1);
  });

  it("renders children - message controls", () => {
    const messageReadControlls = findByTestAttr(
      wrapper,
      "message-read-controlls"
    );
    expect(messageReadControlls.length).toBe(1);
  });

  it("renders fallback when message with that ID is not found", () => {
    const initialState = {
      lang: {
        languagePanelHidden: true,
        currentLanguage: "eng",
      },
      messages: {
        allMessages: {},
        isLoading: false,
        errorMessage: null,
      },
    };
    const store = storeFactory(initialState);
    wrapper = setShallowWRC(<MessageRead store={store} />);

    const messageReadFallback = findByTestAttr(
      wrapper,
      "message-read-fallback"
    );
    expect(messageReadFallback.length).toBe(1);
  });

  it("matches snapshot", () => {
    const messageRead = findByTestAttr(wrapper, "message-read");
    expect(messageRead).toMatchSnapshot();
  });
});
