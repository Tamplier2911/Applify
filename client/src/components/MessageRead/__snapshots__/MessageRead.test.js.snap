// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageRead /> matches snapshot 1`] = `
<styled.div
  data-test="message-read"
>
  <styled.div>
    <styled.div>
      <styled.div>
        Test Message
      </styled.div>
      <styled.div>
        test@email.com
      </styled.div>
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.img
          alt="happy user"
          src="http://localhost/api/uploads/images/users/default.jpg"
        />
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    data-test="message-read-body"
  >
    <styled.div>
      To be tested!
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      May 31, 2020
    </styled.div>
  </styled.div>
  <styled.div
    data-test="message-read-controlls"
  >
    <Button
      click={[Function]}
      data-test="message-read-delete"
      type="button"
      value="Delete"
    />
    <Button
      click={[Function]}
      data-test="message-read-email"
      type="button"
      value="Answer"
    />
  </styled.div>
  <withRouter(GetBack)
    path="/profile/messages"
  />
</styled.div>
`;
